tests/DataValues/MonolingualTextValueTest.php

```diff
@@ -27,44 +27,66 @@ public function getClass() {
 	}
 
 	public function validConstructorArgumentsProvider() {
-		$argLists = [];
+		return [
+			[ 'en', 'foo' ],
+			[ 'en', ' foo bar baz foo bar baz foo bar baz foo bar baz foo bar baz foo bar baz ' ],
+		];
+	}
 
-		$argLists[] = [ 'en', 'foo' ];
-		$argLists[] = [ 'en', ' foo bar baz foo bar baz foo bar baz foo bar baz foo bar baz foo bar baz ' ];
+	public function invalidConstructorArgumentsProvider() {
+		return [
+			[ 42, null ],
+			[ [], null ],
+			[ false, null ],
+			[ true, null ],
+			[ null, null ],
+			[ 'en', 42 ],
+			[ 'en', false ],
+			[ 'en', [] ],
+			[ 'en', null ],
+			[ '', 'foo' ],
+		];
+	}
 
-		return $argLists;
+	public function testNewFromArray() {
+		$array = [ 'text' => 'foo', 'language' => 'en' ];
+		$value = MonolingualTextValue::newFromArray( $array );
+		$this->assertSame( $array, $value->getArrayValue() );
 	}
 
-	public function invalidConstructorArgumentsProvider() {
-		$argLists = [];
+	/**
+	 * @dataProvider invalidArrayProvider
+	 */
+	public function testNewFromArrayWithInvalidArray( array $array ) {
+		$this->setExpectedException( 'DataValues\IllegalValueException' );
+		MonolingualTextValue::newFromArray( $array );
+	}
 
-		$argLists[] = [ 42, null ];
-		$argLists[] = [ [], null ];
-		$argLists[] = [ false, null ];
-		$argLists[] = [ true, null ];
-		$argLists[] = [ null, null ];
-		$argLists[] = [ 'en', 42 ];
-		$argLists[] = [ 'en', false ];
-		$argLists[] = [ 'en', [] ];
-		$argLists[] = [ 'en', null ];
-		$argLists[] = [ '', 'foo' ];
+	public function invalidArrayProvider() {
+		return [
+			[ [] ],
+			[ [ null ] ],
+			[ [ '' ] ],
+			[ [ 'en', 'foo' ] ],
+			[ [ 'language' => 'en' ] ],
+			[ [ 'text' => 'foo' ] ],
+		];
```

Comment:
It's nicer to use `yield` btw

Reply:
I'm curious. How would this look like in a test case provider?

Reply:
yield [ 1 ];
yield [ 2 ];
yield 'such case' =>  [ 3 ];